search-app/src/cmr/search/services/query_service.clj

```diff
@@ -336,32 +338,28 @@
       (map :concept-id (:items results)))))
 
 (defn- get-highest-visible-revisions
-  "Returns the query and the highest visible collection revisions search result of the given
-   collection concept ids in the given result format."
-  [context coll-concept-ids result-format]
-  (when (seq coll-concept-ids)
-    ;; find all collection revisions, then filter out the highest revisions
-    ;; and replace the hits and items in results with those of the highest revisions.
-    (let [condition (gc/and-conds
-                     [(qm/string-conditions :concept-id coll-concept-ids true)
-                      (qm/boolean-condition :deleted false)])
-          query (qm/query {:concept-type :collection
-                           :condition condition
-                           :all-revisions? true
-                           :page-size :unlimited
-                           :result-format result-format})
-          results (qe/execute-query context query)
-          highest-coll-revisions (u/map-values
-                                  #(apply max (map :revision-id %))
-                                  (group-by :concept-id (:items results)))
-          highest-revisions (filter
-                             (fn [coll]
-                               ((set highest-coll-revisions)
-                                [(:concept-id coll) (:revision-id coll)]))
-                             (:items results))]
-      (-> results
-          (assoc :items highest-revisions)
-          (assoc :hits (count highest-revisions))))))
+  "Returns the query and the highest visible collection revisions search result
```

Comment:
I was going to duplicate a ton of logic, so I just refactored it slightly to take a query-condition, and allow us both to use it